packages/edit-site/src/components/post-fields/index.js

```diff
@@ -369,6 +370,13 @@ function usePostFields( viewType ) {
 					return <time>{ getFormattedDate( item.date ) }</time>;
 				},
 			},
+			{
```

Comment:
This works because our APIs are already extensible. However, there's an alternative that @joshuatf mentioned somewhere. Instead of doing:

```js
{
  id: 'slug',
  Edit: /* custom slug Edit */,
  render: /* custom slug render */,
  isValid: /* custom slug isValid */,
  // etc.
}
```

we can do something along these lines:

```js
registerFieldType( 'core/slug', FieldSlug );

const fields = [
  {
    id: 'slug',
    type: 'core/slug',
  }
];
```

This has a few advantages:

- It gets us a bit closer to making the `type` field required. We're not there there yet because there are a couple of missing types we need (image, for example), but we should aim to do it.

- We offer higher semantics and gets us a bit closer to having no-code JSON-oriented fields. This may sound a bit abstract, but imagine the following scenario: in the future, there's a mechanism for users to create screens based on declaring which fields they want from a dataset. All via UI. In that scenario, that UI needs to be no-code, and users shouldn't have to provide a "custom render", "custom edit", "custom validation", etc. if they just want a field to be a "slug". Instead, they should work with types: the basic ones provided by DataViews (string, integer, date, etc.) but also any other one that is domain-specific (slug, cart-status, etc.).

Another aspect of this PR is what should be the API offered by the `wordpress/field` package?

This suggest it should export individual field properties (render, edit, isValid, etc.). There are a few. What if it exported the whole field instead? An advantage of exporting the whole Field is that the `wordpress/field` package can implement new APIs as they come without having to modify the consumers:

```js
// The consumer doesn't have to modify any code when
// a new Field property is introduced, renamed it, etc.
// It's the FieldSlug responsibility to implement it.

const fields = [
  {
    id: 'slug',
    type: 'core/slug'
  }
];
```